### PR TITLE
Use NNC support from CpGrid.

### PR DESCRIPTION
**Pull request by atgeirr**
With this, arbitrary NNCs should work. No testing yet, so still work in progress. Requires OPM/opm-grid#355.

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this opm-grid=355 please

**Comment by blattms** [comment]
Please see https://github.com/OPM/opm-grid/pull/355/files#r251450717 for a possible explanation why the test fails.

**Comment by atgeirr** [comment]
jenkins build this opm-grid=355 please

File: ebos/ecltransmissibility.hh [hunk 1]
```diff
@@ -277,6 +277,14 @@ public:
                 unsigned insideFaceIdx  = intersection.indexInInside();
                 unsigned outsideFaceIdx = intersection.indexInOutside();
 
+                if (insideFaceIdx == -1) {
+                    // NNC. Set zero transmissibility, as it will be
```
[the remark this answers is not in the document]
Comment (andlaus) [reply ebos/ecltransmissibility.hh]: in this case the warning is not even a false positive. Thanks for bringing this up!

(if the `{in,out}sideFaceIdx` are changed to signed integers the compiler will probably produce some other signedness warnings when using @atgeirr's warning flags.)

**Comment by blattms** [comment]
@akva2, is mpi.compareECLInitFiles_flow+NORNE_ATW2013 a serial run with a binary compiled with MPI support or a real parallel run?

**Comment by akva2** [comment]
all tests named compareECLInitFiles is serial. the mpi in front is only the build configuration, ie, we have built with mpi enabled. only the compareParallelSim and compareParallelRestartedSim involves parallel runs.

**Comment by atgeirr** [comment]
I have to think about the failed test a little, I am not sure if it matters. But I struggle to see why TRANX should change.

Apart from that, the test case provided by @totto82 works fine with the current PRs.

**Comment by blattms** [comment]
Could be some memory issues that only kick in with MPI. Have you tried running your small tests with valgrind?

**Comment by atgeirr** [comment]
jenkins build this opm-grid=355 please

**Comment by atgeirr** [comment]
jenkins build this opm-grid=355 please

**Comment by alfbr** [comment]
benchmark opm-grid=355

**Pull request merged**